### DISCUSSION TITLE
How to set a pure white background for a Skybox?

### DISCUSSION THREAD

**Discussion by linconz** [google/filament#7992] [Q&A]
Hello everyone, I am currently using the iOS demo code hello-gltf. I have modified the `setupIbl` function in `App.cpp`. Here is the modified code:
```
void App::setupIbl() {
    
    const uint32_t textureSize = 1;
    const uint8_t whitePixel[] = { 255, 255, 255, 255 };

    filament::Texture::PixelBufferDescriptor buffer(whitePixel, sizeof(whitePixel),
        filament::Texture::Format::RGBA, filament::Texture::Type::UBYTE);

    filament::Texture *whiteCubeTexture = filament::Texture::Builder()
        .width(textureSize)
        .height(textureSize)
        .levels(1)
        .sampler(filament::Texture::Sampler::SAMPLER_CUBEMAP)
        .format(filament::Texture::InternalFormat::RGBA8)
        .build(*engine);

    for (int face = 0; face < 6; face++) {
        filament::Texture::PixelBufferDescriptor faceBuffer(whitePixel, sizeof(whitePixel),
            filament::Texture::Format::RGBA, filament::Texture::Type::UBYTE);
        whiteCubeTexture->setImage(*engine, 0, 0, 0, face, textureSize, textureSize, 1, std::move(faceBuffer));
    }

    // white cube texture
    app.skyboxTexture = whiteCubeTexture;
    app.skybox = filament::Skybox::Builder()
        .environment(app.skyboxTexture)
        .build(*engine);

    scene->setSkybox(app.skybox);

    // sun light
    app.sun = EntityManager::get().create();
    LightManager::Builder(LightManager::Type::SUN)
        .castShadows(true)
        .direction({0.0, -1.0, 0.0})
        .build(*engine, app.sun);
    scene->addEntity(app.sun);
}
```

Question 1:
I just want to a pure white `Skybox`, but it currently appears black. Why is this happening?

Question 2:
In the Android demo code, I see a class like HDRLoader that can load HDR files directly without the need for pre-compilation. However, on iOS, it seems that HDR resources must be pre-compiled using Xcode, which is very unfriendly. How can I directly load and use HDR files in iOS like in the Android code?
Because on the design resources side, HDR files can only be delivered via RESTful API access.

![截屏 2024-07-23 22 28 47](https://github.com/user-attachments/assets/53b07efb-e8ad-4131-be04-f68c9b4f087b)

**Comment by CloseFile** [comment]
I do not know how it works on iOS. But on Android, to get a pure white color, I've changed the ToneMapper and experimented with the settings. 
//        view.colorGrading = ColorGrading.Builder()
//            .toneMapper(ToneMapper.Generic(2.0f, 0.25f, 0.3f, 20.0f))
//            .build(engine)

**Comment by mbalajee** [comment]
Again an android answer, simply setting skybox to white works for me in Android
`scene.skybox = Skybox.Builder().color(1f, 1f, 1f, 1f).build(engine)`

**Reply by drandro7** [reply]
@mbalajee I see that you posted a color that is black. I set the color to 1,1,1 but it still seems it is beige in color instead of white.
see screenshot from an example project: 
![image](https://github.com/user-attachments/assets/0714f444-91a5-44df-9856-2361f84ced4f)

**Reply by drandro7** [reply]
`modelViewer.scene.skybox = Skybox.Builder().color(1.0f, 1.0f, 1.0f, 1.0f).build(modelViewer.engine)`

**Reply by mbalajee** [reply]
This doesn't work for you?
https://github.com/google/filament/blob/fa0e20a6997d6034f4a37c8347360c7cbd5421e0/android/samples/sample-hello-triangle/src/main/java/com/google/android/filament/hellotriangle/MainActivity.kt#L123
